driver-core/src/test/java/com/datastax/driver/core/ProtocolBetaVersionTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import org.testng.annotations.Test;
+
+import static com.datastax.driver.core.ProtocolVersion.V4;
+import static com.datastax.driver.core.ProtocolVersion.V5;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+/**
+ * Tests for the new USE_BETA flag introduced in protocol v5
+ * and Cassandra 3.10.
+ */
+@CCMConfig(version = "3.10")
+public class ProtocolBetaVersionTest extends CCMTestsSupport {
+
+    /**
+     * Verifies that the driver can connect to 3.10 with the following combination of options:
+     * Version V5
+     * Flag SET
+     * Expected version: V5
+     *
+     * @jira_ticket JAVA-1248
+     */
+    @Test(groups = "short")
+    public void should_connect_with_beta_when_beta_version_explicitly_required_and_flag_set() throws Exception {
+        Cluster cluster = Cluster.builder()
+                .addContactPoints(getContactPoints())
+                .withPort(ccm().getBinaryPort())
+                .allowBetaProtocolVersion()
+                .build();
+        cluster.connect();
```

Comment:
I see, it's because [Protocol V5 expects](https://github.com/apache/cassandra/blob/cassandra-3.11/doc/native_protocol_v5.spec#L1183-L1184) unsigned int for query options flags [QueryOptions code](https://github.com/apache/cassandra/blob/cassandra-3.11/src/java/org/apache/cassandra/cql3/QueryOptions.java#L405)  (see also: [PYTHON-684](https://datastax-oss.atlassian.net/browse/PYTHON-684)).  I've fixed it locally.  I guess this is the nature of using beta to begin with!  I'm not sure we need to fix int his PR, but here's a gist of the patch to get things working.